algorithms/scripts/SGDClassifier-master.py

```diff
@@ -0,0 +1,309 @@
+
+# coding: utf-8
+
+# # Create a logistic regression model to predict TP53 mutation from gene expression data in TCGA
+
+# In[1]:
+
+import os
+import urllib
+import random
+import warnings
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing, grid_search
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+
+
+# In[2]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[3]:
+
+# We're going to be building a 'TP53' classifier 
+GENE = 'TP53'
+
+
+# In[4]:
+
+# Parameter Sweep for Hyperparameters
+n_feature_kept = 500
```

Comment:
@gwaygenomics I agree with you. For performance comparison, I think we should use 8000 (see the graph [here](https://github.com/cognoma/machine-learning/issues/22#issuecomment-238113032)) or may be just keep all features and let everyone pick a combo of feature selection/classification (e.g.. recursive feature elimination + SVM). Though I'm not familiar with gene classification, intuitively feature selection would be crucial here consider we have 20000+ features but only 7000+ samples so I think feature selection should be included as a part of the algorithms  to be tested